gnbsim.go

```diff
@@ -62,65 +64,37 @@ func action(c *cli.Context) error {
 	logger.AppLog.Infoln("Setting log level to:", lvl)
 	logger.SetLogLevel(lvl)
 
-	profile.InitializeAllProfiles()
+	prof.InitializeAllProfiles()
 	err := gnodeb.InitializeAllGnbs()
 	if err != nil {
 		logger.AppLog.Errorln("Failed to initialize gNodeBs:", err)
 		return err
 	}
 
-	summaryChan := make(chan common.InterfaceMessage)
-	result := "PASS"
+	go ListenAndLogSummary()
 
 	var profileWaitGrp sync.WaitGroup
-	for _, profileCtx := range config.Configuration.Profiles {
-		if !profileCtx.Enable {
+	for _, profile := range config.Configuration.Profiles {
+		if !profile.Enable {
 			continue
 		}
 		profileWaitGrp.Add(1)
 
 		go func(profileCtx *profctx.Profile) {
 			defer profileWaitGrp.Done()
+			go prof.ExecuteProfile(profileCtx, profctx.SummaryChan)
+		}(profile)
 
-			go profile.ExecuteProfile(profileCtx, summaryChan)
-
-			select {
-			// Waiting for execution summary from profile routine
-			case m, ok := <-summaryChan:
-				if !ok {
-					logger.AppLog.Fatalln("summary Channel closed")
-					break
-				}
-				msg, ok := (m).(*common.SummaryMessage)
-				if !ok {
-					logger.AppLog.Fatalln("Invalid message type")
-					break
-				}
-
-				logger.AppSummaryLog.Infoln("Profile Name:", msg.ProfileName, ", Profile Type:", msg.ProfileType)
-				logger.AppSummaryLog.Infoln("Ue's Passed:", msg.UePassedCount, ", Ue's Failed:", msg.UeFailedCount)
-
-				if msg.UeFailedCount != 0 {
-					result = "FAIL"
-				}
-
-				if len(msg.ErrorList) != 0 {
-					logger.AppSummaryLog.Infoln("Profile Errors:")
-					for _, err := range msg.ErrorList {
-						logger.AppSummaryLog.Errorln(err)
-					}
-				}
-			}
-		}(profileCtx)
 		if config.Configuration.ExecInParallel == false {
 			profileWaitGrp.Wait()
 		}
 	}
+
 	if config.Configuration.ExecInParallel == true {
 		profileWaitGrp.Wait()
 	}
 
-	logger.AppSummaryLog.Infoln("Simulation Result:", result)
+	httpserver.StartHttpServer()
```

Comment:
This is blocking call. Also currently there is not wait group maintained for profiles executed through APIs

Reply:
First all the locally configured profiles are executed

Reply:
Since profiles can execute indepedently, you can add separate go routing for http server. Now question comes is what to do when all local profiles are executed then how long we want to wait for any profile to be received from API. Run the profile http server only if configuration exist else dont even wait.